### PR TITLE
Add pre-commit/post-merge hooks

### LINKED ISSUE #15: Create pre-commit hook
Travis is protecting our main branches but it’s not officially documented what our build is doing, nor is it very convenient to run all those commands locally. So, we should add a build script to run all those commands and have contributors add this as a pre-commit hook, so that we don’t have to run a Travis build to find linting errors.

### PR DESCRIPTION
**Pull request by rwblickhan**
# Pull Request

## Description

Give a brief description of your changes:

Adds pre-commit and post-merge hooks and a Makefile to install them, so that dependencies are kept up to date and we're less likely to commit something that will fail the Travis build.

## Ticket(s)

Closes #15

### CODE REVIEW

File: scripts/build_check.sh [hunk 1]
```diff
@@ -0,0 +1,6 @@
+#!/usr/bin/env bash
+set -euxo pipefail
+pipenv run pycodestyle .
+pipenv run pydocstyle .
+mdl .
```
Comment (cheukyin699) [inline scripts/build_check.sh]: Maybe switch to using `bundle exec mdl` with `Gemfile`? Though this raises the complexity a tad bit (with the extra 1-line-long file).

Or we could remove it entirely and switch to the unmaintained python one.

I personally like the latter.

File: README.md [hunk 2]
```diff
@@ -32,4 +32,12 @@ environment like so:
 pipenv shell
 ```
 
-and then commands like `pycodestyle` and `pytest` can be run like normal.
\ No newline at end of file
+and then commands like `pycodestyle` and `pytest` can be run like normal.
+
+Additionally, we use [Travis CI](https://travis-ci.org/ubclaunchpad/rocket2.0) as
+a CI system. To run the same checks locally, please install our git hooks:
```
Comment (cheukyin699) [inline README.md]: In addition to installer git hooks, can also mention that user/dev can run the scripts in the folder to achieve the same effect, without committing/merging.

File: scripts/build_check.sh (hunk 1 above)
Comment (cheukyin699) [reply scripts/build_check.sh]: Never mind. I think it's just that my ruby/gem installation is busted. Resolving.

Comment (rwblickhan) [reply scripts/build_check.sh]: Hmm this does seem problematic. I'm getting issues with `gem install ...` as well (I have to use sudo) but then installing bundler requires... `gem install bundler`. Perhaps we should just come back to this later?

**Pull request merged**